### PR TITLE
Add Github CI workflow

### PR DESCRIPTION
**Pull request by jku**
### Description of the changes being introduced by the pull request:

Because of recent issues with travis I ended up polishing my Github CI commit a bit:
* tests enabled on linux and mac
* windows is disabled due to gpg hanging and a storage test failure: these need to be investigated
* The no-gpg cases are now handled using the new environment variable
* No coveralls support

### CODE REVIEW

**Comment by jku** [comment]
The Windows gpg failures look like this 
```
TimeoutExpired: Command '['gpg', '--detach-sign', '--digest-algo', 'SHA256', '--homedir', 'c:/users/runner~1/appdata/local/temp/tmprxgwki/rsa']' timed out after 3 seconds
```
And the storage test failure:

```
======================================================================
FAIL: test_exceptions (tests.test_storage.TestStorage)
----------------------------------------------------------------------
Traceback (most recent call last):
File "D:\a\securesystemslib\securesystemslib\tests\test_storage.py", line 66, in test_exceptions
    self.storage_backend.create_folder, '/none/existent/path')
 AssertionError: StorageError not raised
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          # A match with 'restore-keys' is used as fallback
+          restore-keys: ${{ runner.os }}-pip-
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install --upgrade tox
+
+      - name: Run tox
+        # See TOXENV environment variable for the testenv to be executed here
```
Comment (jku) [inline .github/workflows/ci.yml:61]: oops this is a leftover comment

File: tox.ini [hunk 2]
```diff
@@ -4,7 +4,7 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py27, py36, py37, py38, py39 purepy27, purepy38
+envlist = py27, py36, py37, py38, py39, purepy27, purepy38, py27-no-gpg, py38-no-gpg
```
Comment (lukpueh) [inline tox.ini:7]: Good catch! Interesting how tox successfully builds a `py39purepy27` env. :)

**Review by lukpueh — CHANGES_REQUESTED** [review]
Thanks for adding the workflow file, @jku! It looks quite good to me. We should ticketize the remaining tasks i.e.:
- reinstate coveralls
- enable windows builds

I can do that if you want. 

I also suggest to remove .travis.yml and all traces to it (badge in readme, manifest.in directive, comment in tox.ini). Can you do that?  Oh and maybe add a new GH action badge to the readme?

**Comment by jku** [comment]
I think I finally managed to get the badge right... but we'll see once there are results for master. The PR should good to go. I can file the two issues after lunch (if you haven't at that point).

**Comment by lukpueh** [comment]
@jku, I added the two tickets #325 and #326, and also took the liberty to push a commit here that removes the coveralls badge (until we fix #325).

**Comment by jku** [comment]
Damn it, almost got the workflow link correct on the badge...

**Comment by lukpueh** [comment]
> Damn it, almost got the workflow link correct on the badge...

Can you send a fix?

**Pull request merged**